lib/internal/streams/readable.js

```diff
@@ -367,6 +367,18 @@ Readable.prototype.push = function(chunk, encoding) {
   debug('push', chunk);
 
   const state = this._readableState;
+  state[kState] &= ~kReading;
```

Comment:
This changes the behavior.

before, if the user tried to push a number to stream in byte mode, we would set the reading to "false" (unsettling the `kReading` byte

now we do...

I'm not sure we want that, @ronag WDYT?

Reply:
Yea... we shouldn't change any state here unless we use the chunk

Reply:
Yes, the `test-webstreams-compose.js` test failing because:
in `push` you moved:
```js
  if ((state[kState] & kEnded) !== 0) {
    errorOrDestroy(stream, new ERR_STREAM_PUSH_AFTER_EOF());
    return false;
  }

  if ((state[kState] & (kDestroyed | kErrored)) !== 0) {
    return false;
  }
```

to be before:
```js
if (!chunk || chunk.length <= 0) {
    state[kState] &= ~kReading;
    maybeReadMore(stream, state);

    return canPushMore(state);
  }
```

Reply:
From my understanding of data flow inside Readable stream, this change would be safe. I was wrong. All breaking changes are already reverted.